.github/workflows/ci.yaml

```diff
@@ -0,0 +1,186 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+          cache: npm
+
+      - name: Install dependencies
+        run: npm ci --ignore-scripts
+
+      - name: Build
+        run: npm exec -- lerna run build --concurrency=2
+
+      - name: Upload dist artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: dist-artifact
+          path: packages/**/dist
+
+  build-demo:
```

Comment:
I'd like to see this folded directly into `deploy`. We're not getting a ton of value here, and the new workflow could be simplified with:

- build
  - lint
  - test
  - deploy
    - publish

where (lint, test, deploy) run in parallel. Furthermore, it makes good sense to wait for a successful `build` before doing the `deploy`.

Reply:
Looks great and it's fast! 🔥 🚀 

![Screenshot 2024-06-18 at 11 48 04 AM](https://github.com/zendeskgarden/react-components/assets/6879688/0f9dd72c-5680-4367-a4d4-2a1930416293)

On `main`, do we still want to deploy even if `test`& `lint` have not passed?

Reply:
> On main, do we still want to deploy even if test& lint have not passed?

Hmm, good Q. No I don't think we do.